src/components/CourseModuleList.vue

```diff
@@ -1,16 +1,16 @@
 <script setup>
 
-import { useCoursesStore } from '../stores/courses';
 import { useRoute } from 'vue-router';
+import { useCoursesStore } from '../stores/courses';
+import courseData from '../assets/data/courses.json';
 
-const coursesStore = useCoursesStore();
 const route = useRoute();
 
-const foundCourse = coursesStore.getCourseById(route.params.courseId);
-
-const moduleList = coursesStore.getModulesList(route.params.courseId);
-
+const courseId = parseInt(route.params.courseId);
+const coursesStore = useCoursesStore(courseData[0], courseId)();
```

Comment:
Create our store. Note the "special" syntax; we're first passing in the requisite args (`courseData` [here we're using `[0]` because our API/business logic shouldn't have any knowledge of this being a singleton] and `courseId`) which _returns_ the `defineStore` function, and then we're invoking that function with `()` to get us the store.